lib/admin/admin.dart

```diff
@@ -1,91 +1,65 @@
+import 'package:client/admin/modify-tutorial/modify-tutorial.dart';
+import 'package:client/admin/update-list-service.dart';
 import 'package:client/service/trylinks_service.dart';
-import 'package:codemirror/codemirror.dart';
 
 @Component(
   selector: 'admin',
   templateUrl: 'admin.html',
-  styleUrls: const ['admin.css'],
-  providers: const [overlayBindings],
+  styleUrls: const ['admin.css', 'package:angular_components/app_layout/layout.scss.css'],
   directives: const [
-    materialDirectives
+    materialDirectives,
+    ROUTER_DIRECTIVES,
+    NgFor
+])
+@RouteConfig(const [
+  const Route(
+    path: '/add-tutorial',
+    name: 'AddTutorial',
+    component: AddTutorialComponent,
+    useAsDefault: true),
+  const Route(
+    path: '/modify-tutorial/:id',
+    name: 'ModifyTutorial',
+    component: ModifyTutorialComponent
+  )
 ])
 
 class AdminPageComponent implements OnInit {
-  String title;
-  CodeMirror descEditor;
-  CodeMirror sourceEditor;
-  String errorMsg;
-  bool confirmTutorialCreation = false;
-  bool validatedInput = false;
-  bool success;
+  List headers;
 
   TryLinksService _service;
   Router _router;
+  UpdateListService _updateListService;
 
-  AdminPageComponent(this._router, this._service);
+  AdminPageComponent(this._router, this._service, this._updateListService);
 
   @override
   Future ngOnInit() async {
     if (!_service.isAdmin()) {
       _router.navigate(['Dashboard']);
     }
+    this.headers = await _service.getTutorialHeaders();
 
-    success = null;
-
-    Map descOptions = {
-      'mode': 'markdown',
-      'autofocus': true,
-      'lineWrapping': true,
-      'indentWithTabs': true,
-    };
-
-    Map sourceOptions = {
-      'mode': 'links',
-      'theme': 'material',
-      'lineNumbers': true,
-      'lineWrapping': true,
-      'indentWithTabs': true,
-    };
-
-    this.descEditor = new CodeMirror.fromTextArea(
-        querySelector('textarea.new-tutorial-desc'),
-        options: descOptions);
-    this.descEditor.setSize('100%', '77vh');
-
-    this.sourceEditor = new CodeMirror.fromTextArea(
-        querySelector('textarea.new-tutorial-editor'),
-        options: sourceOptions);
-    this.sourceEditor.setSize('100%', '77vh');
+    _updateListService.updateList.listen((data) =>
+      _service.getTutorialHeaders()
+        .then((headers) => this.headers = headers)
+        .catchError(() => print("Could not retrieve the list of tutorials"))
+    );
   }
 
-  Future onCreateTutorial() async {
-    final result = await _service.createTutorial(
-        title, this.descEditor.getDoc().getValue(), this.sourceEditor.getDoc().getValue());
-
-    if (result != true) {
-      this.success = false;
-      print('Failed to create a new tutorial');
-    } else {
-      this.success = true;
-      this.title = null;
-      this.descEditor.getDoc().setValue('');
-      this.sourceEditor.getDoc().setValue('');
-    }
+  void updateList() {
```

Comment:
What's this function for?

Reply:
Right, I used it when experimenting with the shared service between the Admin Components and its children, and then forgot to delete it.